components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceManagementServiceImpl.java

```diff
@@ -301,4 +304,48 @@ public Response getEffectivePolicyOfDevice(@PathParam("type") String type,
         }
     }
 
+    @GET
+    @Path("{type}/{id}/compliance-data")
+    public Response getComplianceDataOfDevice(@PathParam("type") String type,
+                                              @PathParam("id") String id) {
+
+        RequestValidationUtil.validateDeviceIdentifier(type, id);
+        PolicyManagerService policyManagementService = DeviceMgtAPIUtils.getPolicyManagementService();
+        Policy policy;
+        ComplianceData complianceData = null;
+        DeviceCompliance deviceCompliance = new DeviceCompliance();
+
+        try {
+            policy = policyManagementService.getAppliedPolicyToDevice(new DeviceIdentifier(id, type));
+        } catch (PolicyManagementException e) {
+            String msg = "Error occurred while retrieving the current policy associated with the '" + type +
+                    "' device, which carries the id '" + id + "'";
+            log.error(msg, e);
+            return Response.serverError().entity(
+                    new ErrorResponse.ErrorResponseBuilder().setMessage(msg).build()).build();
+        }
+
+        if (policy == null) {
+            deviceCompliance.setDeviceID(Integer.valueOf(id));
+            deviceCompliance.setComplianceData(null);
+            //deviceCompliance.setCode(0001l); //code 0001 means no compliance data related to the device
```

Comment:
Yes. Thanks for pointing out. Corrected with following PR.
https://github.com/wso2/carbon-device-mgt/pull/734